wgsl/index.bs

```diff
@@ -3849,109 +4062,147 @@ Note: Override expressions are a superset of [=const-expressions=].
-    within the variable's scope.
-
-Two variables with overlapping lifetimes will not have [=overlap|overlapping memory=].
-When a variable's lifetime ends, its memory may be used for another variable.
-
-When a variable is created, its memory contains an initial value as follows:
-
-* For variables in the [=address spaces/private=] or [=address spaces/function=] address spaces:
-    * The [=zero value=] for the store type, if the variable declaration has no initializer.
-    * Otherwise, it is the result of evaluating the initializer expression at that point in the program execution.
-* For variables in the [=address spaces/workgroup=] address space:
-    * When the store type is [=constructible=], the [=zero value=] for the store type.
-    * If the store type is an [=atomic type=], the [=zero value=] is that of the underlying type ([=integer scalar=]).
-    * Otherwise, if the store type is not constructible, the [=zero value=] is determined by recursively applying these
-        rules to each component of the [=composite=] until a [=constructible=] type is encountered.
-        * Note: This commonly occurs when using an array with a [=pipeline-overridable=] element count or
-            a composite that contains an atomic type.
-* Variables in other address spaces are [=resources=]
-    set by bindings in the [=draw command=] or [=dispatch command=].
+execution for which the [=memory locations=] are associated with the variable.
+The lifetime of a [=module scope=] variable is the entire execution of the
+shader stage.
+There is an independent version of a variable in the [=address spaces/private=]
+and [=address spaces/function=] address spaces for each invocation.
+[=function scope|Function-scope=] variables are a [=dynamic context=].
+The lifetime of a function-scope variable is determined by its scope:
+* It starts when control enters the variable’s declaration.
+* It ends when the name is no longer [=in scope=] of any part of the [=dynamic context=].
+    That is, the lifetime includes any functions [=function call|called=] while
+    the name is in scope.
+
+Two variables with overlapping lifetimes will not have [=overlap|overlapping
+memory=] locations.
+When a variable’s lifetime ends, its memory may be used for another variable.
+
+Note: WGSL ensures the contents of a variable are only observable during the
+variable’s lifetime.
+
+When a variable in the [=address spaces/private=], [=address spaces/function=],
+or [=address spaces/workgroup=] address spaces is created, it will have an
+initial value.
+If no initializer is specified the initial value is the <dfn noexport>default
+initial value</dfn>.
+The initial values are computed as follows:
+* For variables in the function address space:
+    * The [=zero value=] of the [=store type=], if the variable declaration did
+        not specify an initializer.
+    * Otherwise for function variables, it is the result of evaluating the
```

Comment:
Does `function variables` mean anything here that isn't conveyed by this being in the `function address space` bullet point? Was trying to determine if there was something extra this implied.